front/pages/w/[wId]/u/chat/[cId]/index.tsx

```diff
@@ -518,11 +521,41 @@ export function MessageView({
           </div>
           <div
             className={classNames(
-              "break-word ml-2  flex flex-1 flex-col whitespace-pre-wrap pt-1",
+              "break-word relative  ml-2 flex flex-1 flex-col whitespace-pre-wrap pt-1",
               message.role === "user" ? "italic text-gray-500" : "text-gray-700"
             )}
           >
             {toMarkdown(message)}
+            {message.role === "assistant" ? (
+              <div
+                className="absolute -top-1.5 right-0 mt-2 hidden text-gray-400 hover:text-violet-800 group-hover:block"
+                onClick={async () => {
+                  await navigator.clipboard.writeText(message.message || "");
+                  // make the tooltip relative to this element appear
```

Comment:
I don't like tooltips, they are going on top of other elements and it often leads to some cracra things to do 😄 
Feel free to keep as is of course, I'm just sharing about what I would have done: 
- **Non lazy option** - I would have defined a banner zone in the chat that I can use to confirm things to the user, such as "Copied to clipboard!", "Thanks for the feedback!" and managed it with a useState.
- **Lazy option** - I would have just replaced the icon from `ClipboardDocumentListIcon` to `ClipDocumentCheck` and put it purple 😄 


Tooltip works of course, and the current UX is clear! 
About the code, do you think it's relevant to make it a bit more React? Something more like that (html simplified to ease the reading)? 

```js
const CopyMessageToClipBoard = ({ message }) => {
  const [showTooltip, setShowTooltip] = useState(false);

  const handleButtonClick = () => {
    setShowTooltip(true);
    setTimeout(() => setShowTooltip(false), 3000);
  };

  return (
    <div>
      <span>{message.text}</span>
      <button onClick={handleButtonClick}>Show Tooltip</button>
      {showTooltip && <div className="tooltip">This is a tooltip!</div>}
    </div>
  );
};
```

And you call it with: 

```js
{message.role === "assistant" && message.message && (
    <CopyMessageToClipBoard message={message.message} />
)}
```

Reply:
Yep better that way :+1:

Reply:
Regarding tooltip VS icon vs banner, I think the lazy option is also the right one: people know what the clip is, bannering it is kind of heavy! Will try it and confirm with ed

Reply:
👍🏻